libs/reactive-forms/src/lib/form-array.ts

```diff
@@ -21,14 +21,19 @@ import {
 } from './core';
 import { DeepPartial } from './types';
 
+type ValueOfControl<T> = T extends FormControl<infer C>
```

Comment:
so why this: 
if we infer like in test 
```
    const control = new FormArray([new FormControl<string | null>(null, Validators.required)]);
    const spy = jest.fn();
    control.invalid$.subscribe(spy);
    expect(spy).toHaveBeenCalledWith(true);
    control.setValue(['abc']);
    expect(spy).toHaveBeenCalledWith(false);
    control.setValue([null]);
    expect(spy).toHaveBeenCalledTimes(3);
```
then we will have trouble on line `control.setValue([null]); ` if we use what seems best so 
`readonly value!: ValuesOf<T>[];`
cause in this case `T` is undefined and we only know `Control`,
there will also be a problem if we use  `ValuesOf<Control>[]` if `Control` is `FormControl` cause `ValuesOf` expects `Record`. We cold probably update  `ValuesOf` and udner it `ContolsOf` but I didn't want to make it a too big of a braking change;